force-app/main/default/lwc/newProgramEngagement/newProgramEngagement.js

```diff
@@ -155,6 +179,15 @@ export default class NewProgramEngagement extends LightningElement {
         }
     }
 
+    handleFormChange() {
+        if (!this.hasError) {
+            return;
+        }
+
+        this.form.querySelector("lightning-messages").setError(undefined);
```

Comment:
Do we need to set this to undefined wouldn't record-edit form clear the value when there is no error?

Reply:
No the error does not get cleared until submit is hit again. This will clear the error as soon as the user starts interacting with the form again.